### PR TITLE
[Refactor] Fix stake age checks for regtest

### PR DESCRIPTION
**Pull request by random-zebra**
There is a bug occurring sometimes on regtest where a block is not accepted due to `CheckStakeKernelHash() : min age violation`.
This is because the min age requirement for RegTest was changed (in https://github.com/PIVX-Project/PIVX/pull/812) inside `Stake` but not in `CheckProofOfStake` (the error indicates also the  wrong function).

This fixes it and moves `nStakeMinAge` to chain params (instead of having it as rogue external variable) and sets it to `0` for RegTest (for consistency).
Also fixes the error messages in kernel.cpp.

### CODE REVIEW

**Comment by CaveSpectre11** [comment]
> Also, I'm not a native speaker but as far as I know in English there is no space before a `:` character. So your changes regarding this matter seems to go in the wrong direction.

Personally I don't like the space between the function and `:`.  Can't really judge what the standard format is in the code base since it's pretty much a tossup between the two  (however taking a quick look at `net.cpp` is pretty funny; with `CAddrDB::Write()` and `CBanDB::Write()` having a different format for the same messages.

**Comment by random-zebra** [comment]
Yes, there is no standard convention in the code for that (which is supposed to be just a separator, doesn't have to strictly follow the conventions of the English language for spacing around colons).

Anyway this PR is not about code styling. That can be addressed separately.
The reason for the edits in the log lines was only using `__func__` instead of explicitely writing the name (often mistaking it due to cut and paste).

**Comment by CaveSpectre11** [comment]
> Anyway this PR is not about code styling. That can be addressed separately.

Fair enough; especially after peeking at the code and seeing the topic is a can of worms :)

File: src/kernel.cpp [hunk 1]
```diff
@@ -36,18 +36,18 @@ static std::map<int, unsigned int> mapStakeModifierCheckpoints =
 // Get time weight
 int64_t GetWeight(int64_t nIntervalBeginning, int64_t nIntervalEnd)
 {
-    return nIntervalEnd - nIntervalBeginning - nStakeMinAge;
+    return nIntervalEnd - nIntervalBeginning - Params().StakeMinAge();
```
Comment (CaveSpectre11) [inline src/kernel.cpp]: I went digging to see if there was any division by GetWeight (to see if nStakeMinAge going to zero created a potential division by zero issue), but instead:

Is GetWeight() used anywhere?

Comment (random-zebra) [reply src/kernel.cpp]: Nope. Can be removed.

**Comment by furszy** [comment]
@PIVX-Project/core-developers check this one please, it's pretty straightforward and will clean the #936 commits list for a better review.

**Pull request merged**